relay-common/src/metrics.rs

```diff
@@ -140,15 +178,51 @@ pub fn configure_statsd<A: ToSocketAddrs>(
     prefix: &str,
     host: A,
     default_tags: BTreeMap<String, String>,
+    buffering: bool,
+    sample_rate: f32,
 ) {
     let addrs: Vec<_> = host.to_socket_addrs().unwrap().collect();
     if !addrs.is_empty() {
         log::info!("reporting metrics to statsd at {}", addrs[0]);
     }
-    let statsd_client = StatsdClient::from_udp_host(prefix, &addrs[..]).unwrap();
+
+    let normalized_sample_rate = if sample_rate >= 1.0 {
+        1.0
+    } else if sample_rate <= 0.0 {
+        0.0
+    } else {
+        sample_rate
+    };
+    log::debug!(
+        "metrics sample rate is set to {}{}",
+        normalized_sample_rate,
+        if normalized_sample_rate == 0.0 {
```

Comment:
I don't think clippy likes comparing `==` to floats, and rightly so probably.  Probably should use something like `< f32::MIN_POSITIVE` (or `f64`, whichever this is)

Reply:
apparently clippy is happy, so i'm probably just making stuff up

Reply:
Comparing to `0.0` is exact, and since we don't do arithmetic on the floats, this should be fine.